include/msd/channel.hpp

```diff
@@ -56,10 +64,24 @@ class channel {
      */
     explicit constexpr channel(size_type capacity);
 
+    /**
+     * Sets a timeout for channel operations.
+     *
+     * @param timeout Duration after which operations will time out.
+     */
+    template <typename Rep, typename Period>
+    void setTimeout(const std::chrono::duration<Rep, Period>& timeout);
```

Comment:
Is there a really useful scenario where you want to set a timeout after the channel has been declared, clear the timeout, or change it?

Or would it be enough to have an overloaded constructor that accepts the capacity and the timeout? And then `timeout_` does not need to be atomic. Simplicity is usually better.